chiselFrontend/src/main/scala/Chisel/When.scala

```diff
@@ -2,9 +2,12 @@
 
 package Chisel
 
+import scala.language.experimental.macros
+
 import internal._
 import internal.Builder.pushCommand
 import internal.firrtl._
+import internal.sourceinfo.{SourceInfo, WhenTransform}
 
 object when {  // scalastyle:ignore object.name
```

Comment:
When transforms eliminated, when made final.

Why is this PR overcomplicated? Unless there's a better way to make source locators work with chained apply calls, I think this is a pretty minimal solution that provides source locators while keeping compatibility in all cases.